Doc/tutorial/inputoutput.rst

```diff
@@ -314,6 +314,18 @@ to file data is fine for text files, but will corrupt binary data like that in
 :file:`JPEG` or :file:`EXE` files.  Be very careful to use binary mode when
 reading and writing such files.
 
+If the :func:`open` function is unable to open the file, for instance because
+the file does not exist, it raises an :exc:`OSError` (or one of its
+system-dependent subclasses) which can be handled as follows:
+
+::
+
+   >>> try:
+   ...     f = open('workfile', 'w')
+   ... except OSError as e:
+   ...     print('Cannot open file: ', e)
+   ...
```

Comment:
Is this our recommended way of doing this? Don't we usually recommend using `with open(...) as f:`? I'm thinking that this example, while short, clear and concise, should show the recommended approach.

Reply:
That's a good point. At the same time, I usually prefer try-except blocks to be specific rather than around a large code block like the whole with and everything inside it. What is the recommended way to do open with error handling?

The reason I made this PR was because I felt that the OP of bpo-41737 wasn't clear that the API contract is OSError.  Maybe this is not the way to clarify this point.

Reply:
> What is the recommended way to do open with error handling?

Good question! Maybe the above, and then `with f: ...`? Note that the very next section recommends `with open(...) as f:` as "good practice".

> Maybe this is not the way to clarify this point.

I think this is definitely the right place to show how things should be done, though perhaps after the paragraph introducing using a "with" statement with open().

Reply:
There are some examples here https://docs.python.org/3/tutorial/errors.html